### PR TITLE
Add ability to close App-Menu when user clicks outside

### PR DESCRIPTION
**Pull request by donaldkibet**
### What does this PR do?

1. Add ability to close App-Menu when a user loses its focus

### Screenshoot

screencast 2021-04-01 12-10-59

### CODE REVIEW

File: src/components/navbar-header-panels/app-menu-panel.component.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+interface AppMenuProps {
+  expanded: boolean;
+}
+
+const AppMenuPanel: React.FC<AppMenuProps> = ({ expanded }) => {
+  return (
+    <HeaderPanel
+      aria-label="App Menu Panel"
+      style={{
+        height: "12rem",
```
Comment (FlorianRappl) [inline src/components/navbar-header-panels/app-menu-panel.component.tsx]: I know its set right to 12rem, but I wonder why. Are the designs like this? Why not make it consistent with the other side menu? (user)

File: src/components/navbar/navbar.component.tsx [hunk 2]
```diff
@@ -4,18 +4,18 @@ import UserMenuPanel from "../navbar-header-panels/user-menu-panel.component";
 import SideMenuPanel from "../navbar-header-panels/side-menu-panel.component";
 import Logo from "../logo/logo.component";
 import { navigate, ExtensionSlot } from "@openmrs/esm-framework";
-import AppSwitcher20 from "@carbon/icons-react/lib/app-switcher/20";
+import Switcher20 from "@carbon/icons-react/lib/switcher/20";
```
Comment (FlorianRappl) [inline src/components/navbar/navbar.component.tsx:7]: What's the difference?

**Review by FlorianRappl — COMMENTED** [review]
Looks good. Just a couple of questions.

(Why is it not in the top right corner? There is a bit of a gap and some line if I see that correctly...)

File: src/components/navbar/navbar.component.tsx (hunk 2 above)
Comment (donaldkibet) [reply src/components/navbar/navbar.component.tsx:7]: `AppSwitcher` icon is being deprecated got a console warning for that.

File: src/components/navbar-header-panels/app-menu-panel.component.tsx (hunk 1 above)
Comment (donaldkibet) [reply src/components/navbar-header-panels/app-menu-panel.component.tsx]: made it consistent with the other side menu.

**Comment by donaldkibet** [comment]
> Looks good. Just a couple of questions.
> 
> (Why is it not in the top right corner? There is a bit of a gap and some line if I see that correctly...)

There is no gap or line, maybe the way I recorded the gif didn't capture it well. uploaded a new gif

**Comment by FlorianRappl** [comment]
image

**Comment by donaldkibet** [comment]
The gap was introduced by [MF-430](https://github.com/openmrs/openmrs-esm-primary-navigation/blob/fd38ea8b4569deb27602eda7e9da23deb5d50aca/src/components/navbar/navbar.component.tsx#L123 ) I have removed.

**Review by FlorianRappl — APPROVED** [review]

**Pull request merged**